File: src/main/java/jokatu/components/stores/MessageRepository.java
package jokatu.components.stores;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface MessageRepository {
	@NotNull
	List<Object> getMessageHistory(@NotNull String destination);

	@NotNull
	List<Object> getMessageHistoryForUser(@NotNull String user, @NotNull String destination);

	@Nullable
	Object getLastMessage(@NotNull String destination);

	@NotNull
	Object getLastMessageForUser(@NotNull String user, @NotNull String destination);
}

File: src/main/kotlin/jokatu/components/stores/MessageRepository.kt
package jokatu.components.stores

import jokatu.components.stomp.Destination
import org.springframework.messaging.Message

/**
 * A place whence messages can be retrieved.
 */
interface MessageRepository {
	fun <T: Any> getMessageHistory(destination: Destination<T>): List<Message<T>>

	fun <T: Any> getMessageHistoryForUser(user: String, destination: Destination<T>): List<Message<T>>

	fun <T: Any> getLastMessage(destination: Destination<T>): Message<T>?

	fun <T: Any> getLastMessageForUser(user: String, destination: Destination<T>): Message<T>?
}
